### ISSUE TITLE
how to apply constraints before construct graph?

### ISSUE THREAD

**Issue by bojone** [keras-team/keras#11304]
from the source code, I find the constraints of weights in keras just apply after each gradient descent updating. could I apply it at weights directly?

for example, at Dense layer, kernel_ constraint=unit_norm, can I define x\*unit_norm(W) rather that x\*W?

**Comment by bojone** [comment]
same problem: https://github.com/keras-team/keras/issues/9068
